tools/manifest/tests/test_sourcefile.py

```diff
@@ -542,3 +542,23 @@ def test_no_parse():
 def test_relpath_normalized(input, expected):
     s = create(input, b"")
     assert s.rel_path == expected
+
+
+@pytest.mark.parametrize("url", [b"ref.html",
+                                 b"\x20ref.html",
+                                 b"ref.html\x20",
+                                 b"\x09\x0a\x0c\x0d\x20ref.html\x09\x0a\x0c\x0d\x20"])
+def test_reftest_url_whitespace(url):
+    content = b"<link rel=match href=%s>" % url
```

Comment:
Does this even test this code, rather than the html parser? It seems to check

```html
<link rel=match href= ref.html>
```

rather than an actual attribute with a space.

Reply:
As it happens the HTML parser ignores the spaces there so the test is valid (and hence it passes).